[DDW-633] Implement earned rewards history

daniloprates opened the pull request:
This PR implements detailed earned rewards history.

## Todos

- [ ] Todo

## Screenshots

- Screenshot

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

## Review Checklist

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

gabriela-ponce commented:
@daniloprates @DominikGuzei Here are a few issues and suggestions for the latest build:

1 - The table is not responsive and no scroll is visible. Check [this](https://drive.google.com/file/d/1VqefrQCsta4-vN9GJ4PAj1I4tfjfir3o/view?usp=sharing) screenshot.

2 - Can we have the "Details" button hidden if there are no rewards? Or maybe a message on the table? It's displayed with no data. Check [this](https://drive.google.com/file/d/1dk27s-8acV1vOhw3pqkU8HAUkf8Kz3cn/view?usp=sharing) screenshot.

3 - If you click on the row, the dialog is displayed. Not sure if it's expected, but it looks ok for synced wallets. However, if the wallet is syncing, you can still open the dialog by clicking on the row. Check [this](https://drive.google.com/file/d/1N3ySPpjLbeEGAXyhw8y_bjg25WuoyQkn/view?usp=sharing) video.

4 - I got an API error when trying to see the rewards of a wallet in mainnet. Check [this](https://drive.google.com/file/d/1gP8_PcK7U2Ex0pBtr2TFuQFcrN7spxxu/view?usp=sharing) video.

5 - The name of the csv file doesn't follow the same convention that is used for similar features (FeatureName-WalletName-Timestamp.csv). [This](https://drive.google.com/file/d/1IAR_s0q6fXFmmcAzoTvAL8mFizB5n8Si/view?usp=sharing) is a file exported from the transaction screen and [this](https://drive.google.com/file/d/1MTKaQbrXDbPOrq16nZXEZESvBVK_rm8I/view?usp=sharing) is one from rewards.

johnalotoski commented:
Adding a `hydra-dont-build` label to avoid repeated failed re-eval on hydra until the error is fixed:
```
building '/nix/store/s4dfwmfwmbqc7ky0wwxmpcfbgc0f3cy4-yarn.nix.drv' on 'ssh://root@packet-ipxe-1.ci.iohkdev.io'...
found changes in the lockfile /nix/store/wnvi5rcfwcavb03ifbnhp5qqq4hsb100-yarn.lock
...aborting
builder for '/nix/store/s4dfwmfwmbqc7ky0wwxmpcfbgc0f3cy4-yarn.nix.drv' failed with exit code 1
```
Feel free to remove this label once that build error is being addressed to resume hydra building, thanks.

DominikGuzei commented:
@gabriela-ponce I fixed issue 5 by naming the CSV export file `Rewards-[WALLET_NAME]-[TIME_STAMP]`

daniloprates commented:
Thanks @gabriela-ponce. I've fixed 1-3:

> 1 - The table is not responsive and no scroll is visible. Check [this](https://drive.google.com/file/d/1VqefrQCsta4-vN9GJ4PAj1I4tfjfir3o/view?usp=sharing) screenshot.

Fixed

> 2 - Can we have the "Details" button hidden if there are no rewards? Or maybe a message on the table? It's displayed with no data. Check [this](https://drive.google.com/file/d/1dk27s-8acV1vOhw3pqkU8HAUkf8Kz3cn/view?usp=sharing) screenshot.

Wallets without rewards are not listed, anymore

> 3 - If you click on the row, the dialog is displayed. Not sure if it's expected, but it looks ok for synced wallets. However, if the wallet is syncing, you can still open the dialog by clicking on the row. Check [this](https://drive.google.com/file/d/1N3ySPpjLbeEGAXyhw8y_bjg25WuoyQkn/view?usp=sharing) video.

Syncing wallets are not clickable

> 4 - I got an API error when trying to see the rewards of a wallet in mainnet. Check [this](https://drive.google.com/file/d/1gP8_PcK7U2Ex0pBtr2TFuQFcrN7spxxu/view?usp=sharing) video.
> 5 - The name of the csv file doesn't follow the same convention that is used for similar features (FeatureName-WalletName-Timestamp.csv). [This](https://drive.google.com/file/d/1IAR_s0q6fXFmmcAzoTvAL8mFizB5n8Si/view?usp=sharing) is a file exported from the transaction screen and [this](https://drive.google.com/file/d/1MTKaQbrXDbPOrq16nZXEZESvBVK_rm8I/view?usp=sharing) is one from rewards.

@DominikGuzei fixed 5 and will tackle 4

gabriela-ponce commented:
@daniloprates @DominikGuzei The last fixes look good. Here is a list of new and pending issues on 18255:

**New**:
- Japanese translations are missing, for the [name of the CSV](https://drive.google.com/file/d/1daF54meKcM91ZgHCak1m6n7ZDP4MEn4B/view?usp=sharing), the headers on [the dialog](https://drive.google.com/file/d/1uUKgkWWXE8cGIN2RP8Jcdiz-txgXnq2o/view?usp=sharing) and on [the file](https://drive.google.com/file/d/11iLo7ah_nSL5AioU7pSCJP9z9YnoRLKQ/view?usp=sharing). 

**Pending**:
- I got an API error when trying to see the rewards of a wallet in Mainnet. Check [this](https://drive.google.com/file/d/1gP8_PcK7U2Ex0pBtr2TFuQFcrN7spxxu/view?usp=sharing) video.
- Still no Linux and Windows build available.

daniloprates commented:
> @daniloprates @DominikGuzei The last fixes look good. Here is a list of new and pending issues on 18255:
> 
> **New**:
> 
> * Japanese translations are missing, for the [name of the CSV](https://drive.google.com/file/d/1daF54meKcM91ZgHCak1m6n7ZDP4MEn4B/view?usp=sharing), the headers on [the dialog](https://drive.google.com/file/d/1uUKgkWWXE8cGIN2RP8Jcdiz-txgXnq2o/view?usp=sharing) and on [the file](https://drive.google.com/file/d/11iLo7ah_nSL5AioU7pSCJP9z9YnoRLKQ/view?usp=sharing).
> 
> **Pending**:
> 
> * I got an API error when trying to see the rewards of a wallet in Mainnet. Check [this](https://drive.google.com/file/d/1gP8_PcK7U2Ex0pBtr2TFuQFcrN7spxxu/view?usp=sharing) video.
> * Still no Linux and Windows build available.

@gabriela-ponce I've finished the translation and fixed the Win/Linux builds. @DominikGuzei will check the mainnet issue tomorrow, but meanwhile you can test with Shelley QA, using the wallet Mior found.

gabriela-ponce commented:
@daniloprates The previous fixes look good.

Here are a few issues on the latest build:

1 - The text of the "copy" tooltip for Japanese is displayed in two lines and it looks a little off. Check [this](https://drive.google.com/file/d/12S8rKZHl6d4TMInskbHnLwpkyruxe9TB/view?usp=sharing) screenshot.
2 - The headers for "Epoch" and "Stake Pool" are inverted. Check [this](https://drive.google.com/file/d/1L8jli-41YqBpwX_xQJDX5ObF541OtkHF/view?usp=sharing) screenshot.
3 - (Probably related to the previous issue) Can't order by "Stake pool" column. Check [this](https://drive.google.com/file/d/1h9O8emh9mLqlnIu-VO5p4-1ec_5rHA8w/view?usp=sharing) video.
4 - The amount in the file is not formatted as on Daedalus. Check [this](https://drive.google.com/file/d/1At8qtETExnPXvWixH4LR4YvHLP4gDJsz/view?usp=sharing) file. 
5 - The scrollbar [starts](https://drive.google.com/file/d/1VlZdZ3xzQ182lrDrZCsoRvBCyiBj60vB/view?usp=sharing) on the headers and [doesn't reach the bottom of the table](https://drive.google.com/file/d/1R4cL6yhm5t1CphhAQ3beGdcRWTlI95lI/view?usp=sharing). Can this be improved? I think it should start below the headers and reach the last row when there is no more content.

daniloprates commented:
> @daniloprates The previous fixes look good.
> 
> Here are a few issues on the latest build:
> 
> 1 - The text of the "copy" tooltip for Japanese is displayed in two lines and it looks a little off. Check [this](https://drive.google.com/file/d/12S8rKZHl6d4TMInskbHnLwpkyruxe9TB/view?usp=sharing) screenshot.
> 2 - The headers for "Epoch" and "Stake Pool" are inverted. Check [this](https://drive.google.com/file/d/1L8jli-41YqBpwX_xQJDX5ObF541OtkHF/view?usp=sharing) screenshot.
> 3 - (Probably related to the previous issue) Can't order by "Stake pool" column. Check [this](https://drive.google.com/file/d/1h9O8emh9mLqlnIu-VO5p4-1ec_5rHA8w/view?usp=sharing) video.
> 4 - The amount in the file is not formatted as on Daedalus. Check [this](https://drive.google.com/file/d/1At8qtETExnPXvWixH4LR4YvHLP4gDJsz/view?usp=sharing) file.
> 5 - The scrollbar [starts](https://drive.google.com/file/d/1VlZdZ3xzQ182lrDrZCsoRvBCyiBj60vB/view?usp=sharing) on the headers and [doesn't reach the bottom of the table](https://drive.google.com/file/d/1R4cL6yhm5t1CphhAQ3beGdcRWTlI95lI/view?usp=sharing). Can this be improved? I think it should start below the headers and reach the last row when there is no more content.

Thanks @gabriela-ponce. I've fixed 1-4. Not sure how to fix 5, will try again tomorrow.

gabriela-ponce commented:
@daniloprates I was able to check the first fix, but I got [this API error](https://drive.google.com/file/d/1gOPeJudfr1mDK-HHcUsytJ4bPxbLuO90/view?usp=sharing) on Shelley QA now. It also happens on the previous build where I tested this before.

DominikGuzei commented:
@gabriela-ponce I managed to fix the API query issues for all networks 👍 the 502 error on shelle_qa is not caused by us (and cannot be fixed by us)

gabriela-ponce commented:
@DominikGuzei @daniloprates Here are a few issues/suggestions for build 18313:

- I was able to see rewards for a [Ledger S](https://drive.google.com/file/d/1JrILOlxbBxlWqAt1Z1xpARjBgm_x4tyT/view?usp=sharing) on Mainnet, but no information for a [software wallet](https://drive.google.com/file/d/192b7AfHFV3yrSUvEOS7kguwP-eOn6E2I/view?usp=sharing) on the same network. Both show rewards [on the table](https://drive.google.com/file/d/1RESiHXPNG4Gya-d3VMirs7UirhirSHgG/view?usp=sharing).
- The order in which the items are displayed on the table [is not persisted when exporting the file](https://drive.google.com/file/d/1IFTeIg4NEvoHDIIlFotJZc0ScHSWh2iV/view?usp=sharing). This was the case when exporting on the rewards screen, could this be improved?
- If you open the dialog, cancel when the rewards are loading, and open the dialog again, [the spinner is not displayed](https://drive.google.com/file/d/1inipO96WQxlJzaLCyRM2yLIwS6VTzmrj/view?usp=sharing) (rewards load anyway). 

I was able to check the fixes on [this comment](https://github.com/input-output-hk/daedalus/pull/2572#issuecomment-858099575) (Except # 5) and they look good. I'll create an issue summary on the description.

daniloprates commented:
Thanks @gabriela-ponce. My comments:

> * I was able to see rewards for a [Ledger S](https://drive.google.com/file/d/1JrILOlxbBxlWqAt1Z1xpARjBgm_x4tyT/view?usp=sharing) on Mainnet, but no information for a [software wallet](https://drive.google.com/file/d/192b7AfHFV3yrSUvEOS7kguwP-eOn6E2I/view?usp=sharing) on the same network. Both show rewards [on the table](https://drive.google.com/file/d/1RESiHXPNG4Gya-d3VMirs7UirhirSHgG/view?usp=sharing).

Will leave this to @DominikGuzei 

> * The order in which the items are displayed on the table [is not persisted when exporting the file](https://drive.google.com/file/d/1IFTeIg4NEvoHDIIlFotJZc0ScHSWh2iV/view?usp=sharing). This was the case when exporting on the rewards screen, could this be improved?

This is not desirable. The order in the screen is for viewing only. It's better to be consistent in the spreadsheet exporting. It's easily sortable in the sheet program.

> * If you open the dialog, cancel when the rewards are loading, and open the dialog again, [the spinner is not displayed](https://drive.google.com/file/d/1inipO96WQxlJzaLCyRM2yLIwS6VTzmrj/view?usp=sharing) (rewards load anyway).

Checking this ⌛ 

> I was able to check the fixes on [this comment](https://github.com/input-output-hk/daedalus/pull/2572#issuecomment-858099575) (Except # 5) and they look good. I'll create an issue summary on the description.

I was able to fix # 5. It's pushed

gabriela-ponce commented:
@daniloprates @DominikGuzei On build 18319:

- I was able to check the scroll and it looked ok. However [I'm not getting the list of rewards info from the Ledger S now](https://drive.google.com/file/d/1lvaIHKG6hd-nBTjV_a7RIXaMM5K0y0qj/view?usp=sharing), so I'd need to check again after the fetching is fixed.
- There is no [notification](https://drive.google.com/file/d/1Suvk8rr3UThH81d6s5oLFyoeHwoVvJUW/view?usp=sharing) on the top [when copying the rewards address.](https://drive.google.com/file/d/1lvaIHKG6hd-nBTjV_a7RIXaMM5K0y0qj/view?usp=sharing)

I'll edit the summary list and add a new comment in case I find something else.

daniloprates commented:
> @daniloprates @DominikGuzei On build 18319:
> 
> * I was able to check the scroll and it looked ok. However [I'm not getting the list of rewards info from the Ledger S now](https://drive.google.com/file/d/1lvaIHKG6hd-nBTjV_a7RIXaMM5K0y0qj/view?usp=sharing), so I'd need to check again after the fetching is fixed.
> * There is no [notification](https://drive.google.com/file/d/1Suvk8rr3UThH81d6s5oLFyoeHwoVvJUW/view?usp=sharing) on the top [when copying the rewards address.](https://drive.google.com/file/d/1lvaIHKG6hd-nBTjV_a7RIXaMM5K0y0qj/view?usp=sharing)
> 
> I'll edit the summary list and add a new comment in case I find something else.

@gabriela-ponce fixed the notification issue.

gabriela-ponce commented:
@daniloprates I was able to check the scroll on Testnet and it looks fine.
Also, the notification is displayed now when copying the address.

I updated the issue summary for this PR, but the only pending issues are related to the loading of the data on Mainnet

> I was able to see rewards for a Ledger S on Mainnet, but no information for a software wallet on the same network. Both show rewards on the table.
> 
> If you open the dialog, cancel when the rewards are loading, and open the dialog again, the spinner is not displayed (rewards load anyway).

DominikGuzei commented:
@gabriela-ponce both issues should be fixed now 👍 note: part of the problem is that stake pools take ages to load sometimes, so the current solution is that no stake pool ticker is shown in the rewards history until the stake pools is loaded. I opened a thread in slack to discuss the best strategy for this.

miorsufianiohk commented:
Hi @daniloprates ,

On build 18369, when I clicked "Export", I observed "TypeError: Cannot read property 'ticker'" on console. Also when I went to the folder where the file was saved, the file wasn't there. Please see [video](https://drive.google.com/file/d/1CgOgFPbW41ntmb2l5ujn6UqriHJjk22L/view?usp=sharing)

gabriela-ponce commented:
@DominikGuzei @daniloprates On build 18369:

This issue is fixed:

> If you open the dialog, cancel when the rewards are loading, and open the dialog again, the spinner is not displayed (rewards load anyway).

**New Issues:** 

- I was able to reproduce the same issue @miorsufianiohk reported. Check [this](https://drive.google.com/file/d/1uvq3bGAUW46ROC3YjIAk-2-KE6kbuQtB/view?usp=sharing) screenshot.
- Additionally, if the stake pools are loading, this column on the grid is empty. Check [this](https://drive.google.com/file/d/1i_XcL0PnvbEhzdV8Ka9XgT3Hl0wbJB9s/view?usp=sharing) screenshot. Maybe this option should be disabled until the stake pools load?
- No builds for Windows and Linux.

miorsufianiohk commented:
Hi @daniloprates , on build 18382,

**Pending:**
No builds for Windows and Linux.

**Fixed:**
- Exporting CSV file
- Additionally, if the stake pools are loading, this column on the grid is empty. Check this screenshot. Maybe the "Details" button should be disabled until the stake pools load?

**Question:**
The design and app look different. On design, the reward addresses are truncated in the middle and have copy icons. Also the button text "DETAILS" is capitalised in the design. It's camelcase "Details" in app. Please see [App](https://drive.google.com/file/d/1OBQvNeC25Ab-2-OE1UtxaUnuYD3Bshfq/view?usp=sharing) vs [Design](https://drive.google.com/file/d/16b7XWuOMsLdkb4ngz7_C_FT0dXnIvhLi/view?usp=sharing). cc: @alexander-rukin

alexander-rukin commented:
> **Question:**
> The design and app look different. On design, the reward addresses are truncated in the middle and have copy icons. Also the button text "DETAILS" is capitalised in the design. It's camelcase "Details" in app. Please see [App](https://drive.google.com/file/d/1OBQvNeC25Ab-2-OE1UtxaUnuYD3Bshfq/view?usp=sharing) vs [Design](https://drive.google.com/file/d/16b7XWuOMsLdkb4ngz7_C_FT0dXnIvhLi/view?usp=sharing). cc: @alexander-rukin

"DETAILS" should be capitalised
address should be trimmed, and we actually decided to have copy action only inside dialog, so here in a table it would be just a mention of it (in case of smaller screens - trimmed)

alexander-rukin commented:
<img width="899" alt="image" src="https://user-images.githubusercontent.com/2539345/122917924-39d0a780-d367-11eb-851a-c8f2b637287c.png">
text is not centered vertically

alexander-rukin commented:
<img width="142" alt="image" src="https://user-images.githubusercontent.com/2539345/122922946-d2b5f180-d36c-11eb-861d-6569927dddbe.png">
icon seems bigger than [here](https://zpl.io/2GD3JzE) and not aligned with text properly

alexander-rukin commented:
<img width="162" alt="image" src="https://user-images.githubusercontent.com/2539345/122923028-e9f4df00-d36c-11eb-95be-d736d3a22a05.png">
checkmark is too high

alexander-rukin commented:
<img width="623" alt="image" src="https://user-images.githubusercontent.com/2539345/122923772-b49cc100-d36d-11eb-9b62-f1feff0d26d5.png">

shadows should be shown only on overflow

gabriela-ponce commented:
@daniloprates here are a few issues/suggestions on build 18382:

- The loading state on rewards [shows no message](https://drive.google.com/file/d/1OeIu05EFqnREU_L5MoyteS-4S9kBJANT/view?usp=sharing) informing that this would take a while. Can we include a message over the spinner? Maybe something similar to what we have [on Stake Pools](https://drive.google.com/file/d/1odc5-HFmKlkEZ4ivlc5jnRprkfWThnTG/view?usp=sharing)? 
- Some stake pools are still missing [on the grid](https://drive.google.com/file/d/1oBEYgqxwKABN7GktLH82FYV7e8HepuA1/view?usp=sharing) and [on the document](https://drive.google.com/file/d/1h5tgx6exJ37AryWP-u8RoOQQkV7O5w-3/view?usp=sharing). Can we avoid this? If not, maybe a message that reads "Information unavailable" could be displayed instead? Also, this makes me wonder, what would happen if one of the stake pools on the list is retired?

@alexander-rukin what are your thoughts about these items?

Updating the issue summary with the latest comments ⏳

alexander-rukin commented:
<img width="475" alt="image" src="https://user-images.githubusercontent.com/2539345/122928329-6ccc6880-d372-11eb-8717-46f78a74d135.png">
is it ok to have this top line in grey? and not in a rank color (now we are changing this, but since change is not live...)

alexander-rukin commented:
> * The loading state on rewards [shows no message](https://drive.google.com/file/d/1OeIu05EFqnREU_L5MoyteS-4S9kBJANT/view?usp=sharing) informing that this would take a while. Can we include a message over the spinner? Maybe something similar to what we have [on Stake Pools](https://drive.google.com/file/d/1odc5-HFmKlkEZ4ivlc5jnRprkfWThnTG/view?usp=sharing)?

I think that those title and small text is redundant at this stage -> you just need text + spinner and that's it 

> * Some stake pools are still missing [on the grid](https://drive.google.com/file/d/1oBEYgqxwKABN7GktLH82FYV7e8HepuA1/view?usp=sharing) and [on the document](https://drive.google.com/file/d/1h5tgx6exJ37AryWP-u8RoOQQkV7O5w-3/view?usp=sharing). Can we avoid this? If not, maybe a message that reads "Information unavailable" could be displayed instead? Also, this makes me wonder, what would happen if one of the stake pools on the list is retired?

well, I think we need to wait until everything is loaded

daniloprates commented:
> <img alt="image" width="623" src="https://user-images.githubusercontent.com/2539345/122923772-b49cc100-d36d-11eb-9b62-f1feff0d26d5.png">
> 
> <img alt="image" width="609" src="https://user-images.githubusercontent.com/2539345/122923842-c5e5cd80-d36d-11eb-9485-8f93fdad3734.png">
> 
> shadows should be shown only on overflow

Not possible. It's always or never

daniloprates commented:
@DominikGuzei @gabriela-ponce @alexander-rukin 

I've changed the Rewards list so it'll wait the Stake Pools list to load. I hadn't seen this issue because it loads fast on any other network, but it takes quite a while on mainnet.

Also fixed the styling in the Rewards list. Tomorrow I'll fix the issues in the History dialog.

alexander-rukin commented:
> > <img alt="image" width="623" src="https://user-images.githubusercontent.com/2539345/122923772-b49cc100-d36d-11eb-9b62-f1feff0d26d5.png">
> > <img alt="image" width="609" src="https://user-images.githubusercontent.com/2539345/122923842-c5e5cd80-d36d-11eb-9485-8f93fdad3734.png">
> > shadows should be shown only on overflow
> 
> Not possible. It's always or never

https://lea.verou.me/2012/04/background-attachment-local/

dmitrii-gaico commented:
@daniloprates Hello!
1. Once Rewards Address is copied the pop-up still showing "Copy" it would be great if we could display "Copied" accordingly.
[Record1](https://drive.google.com/file/d/1XkmNQNIHrb1uYk8tB6vK4qk1bHvrQge1/view?usp=sharing)
2. Linux Only. When zooming in the app at some point layout changes even everything still fits just fine.
[Record2](https://drive.google.com/file/d/1Kg42Hcbm2TeFaHHCTd6ZPN8vAtpQrq4_/view?usp=sharing)

cc @alexander-rukin @gabriela-ponce

gabriela-ponce commented:
@daniloprates Here are a few issues on build 18398:

- Can't order the list by the column "Stake pools". This only happens if only stake pool IDs are displayed. Check [this](https://drive.google.com/file/d/1r_IQjq6MArQ_ou29rGBjqpPnodsKqVt7/view?usp=sharing) video. [This](https://drive.google.com/file/d/1JyuhWcxnaJy8p54rAJ1H5F-ET3cpD-In/view?usp=sharing) how it works when the name and ticker are displayed.
- Some missing translations: [CSV on Japanese](https://drive.google.com/file/d/1iGtBewVdY9GOH2yksSkAWC-zMMdX2s7K/view?usp=sharing), [CSV in English](https://drive.google.com/file/d/1WrDT8KyKDhkdDFHQVt1p1AfM4Qh39emP/view?usp=sharing), [tooltip in Japanese](https://drive.google.com/file/d/1VA3NM-fevv4IL7gcPiIc2JX3HeTEj6Yg/view?usp=sharing).

daniloprates commented:
> > > <img alt="image" width="623" src="https://user-images.githubusercontent.com/2539345/122923772-b49cc100-d36d-11eb-9b62-f1feff0d26d5.png">
> > > <img alt="image" width="609" src="https://user-images.githubusercontent.com/2539345/122923842-c5e5cd80-d36d-11eb-9485-8f93fdad3734.png">
> > > shadows should be shown only on overflow
> > 
> > 
> > Not possible. It's always or never
> 
> https://lea.verou.me/2012/04/background-attachment-local/

@alexander-rukin [Hacky and tricky](https://github.com/input-output-hk/daedalus/pull/2572/commits/c7b6b6227ff509cf093789a460468f0d4392b0b1) and it works only on the bottom. Should I keep it?

daniloprates commented:
> @daniloprates Hello!
> 
> 1. Once Rewards Address is copied the pop-up still showing "Copy" it would be great if we could display "Copied" accordingly.
>    [Record1](https://drive.google.com/file/d/1XkmNQNIHrb1uYk8tB6vK4qk1bHvrQge1/view?usp=sharing)
> 2. Linux Only. When zooming in the app at some point layout changes even everything still fits just fine.
>    [Record2](https://drive.google.com/file/d/1Kg42Hcbm2TeFaHHCTd6ZPN8vAtpQrq4_/view?usp=sharing)
> 
> cc @alexander-rukin @gabriela-ponce

@dmitrii-gaico these are fixed, although I wasn't able to reproduce the linux issue

daniloprates commented:
@gabriela-ponce apart from the translation, which is [waiting for Darko's copy](https://docs.google.com/document/d/1R9BIYghgcnaRwKT0EkAv_chVxMAXRX3af9UxbXTNZIE/edit), it's all done.

I've added a new shadow @alexander-rukin, which is a bit unorthodox. It'd be good to test it on all platforms.

dmitrii-gaico commented:
@daniloprates Hello!
Please check the Re-sizing issue I mentioned earlier. Reproducible at Linux/English only. It has a low impact so I believe if it takes a big effort to fix then we can ignore it for now and add it into UI / UX small improvements doc.
[Latest Build](https://drive.google.com/file/d/1fUlL_SPutuPuqz3Q3B6QDAZHZJjVSkJe/view?usp=sharing)

daniloprates commented:
> @daniloprates Hello!
> Please check the Re-sizing issue I mentioned earlier. Reproducible at Linux/English only. It has a low impact so I believe if it takes a big effort to fix then we can ignore it for now and add it into UI / UX small improvements doc.
> [Latest Build](https://drive.google.com/file/d/1fUlL_SPutuPuqz3Q3B6QDAZHZJjVSkJe/view?usp=sharing)

Thanks @dmitrii-gaico. Can you check the latest build?

dmitrii-gaico commented:
> > @daniloprates Hello!
> > Please check the Re-sizing issue I mentioned earlier. Reproducible at Linux/English only. It has a low impact so I believe if it takes a big effort to fix then we can ignore it for now and add it into UI / UX small improvements doc.
> > [Latest Build](https://drive.google.com/file/d/1fUlL_SPutuPuqz3Q3B6QDAZHZJjVSkJe/view?usp=sharing)
> 
> Thanks @dmitrii-gaico. Can you check the latest build?

The record is for 18405 Build - It is the latest one. Currently the 18408 is generated so i'll check it there.

alexander-rukin commented:
<img width="615" alt="image" src="https://user-images.githubusercontent.com/2539345/123411233-e5bd0180-d5b8-11eb-9389-a25cbebadd1c.png">

@daniloprates there is some tiny line in the end 
btw why not possible to do shadow on top? ((

daniloprates commented:
> <img alt="image" width="615" src="https://user-images.githubusercontent.com/2539345/123411233-e5bd0180-d5b8-11eb-9389-a25cbebadd1c.png">
> 
> @daniloprates there is some tiny line in the end
> btw why not possible to do shadow on top? ((

Because it applies background gradient to the element that contains the scrollbar, and it can only be applied once

alexander-rukin commented:
> @alexander-rukin because it applies background gradient to the element that contains the scrollbar, and it can only be applied once
<img width="791" alt="image" src="https://user-images.githubusercontent.com/2539345/123415593-2e2aee00-d5be-11eb-892f-373b6df1b3c3.png">
here it looked like it is possible to have both 
why gradient? and not shadow?

dmitrii-gaico commented:
> > @daniloprates Hello!
> > Please check the Re-sizing issue I mentioned earlier. Reproducible at Linux/English only. It has a low impact so I believe if it takes a big effort to fix then we can ignore it for now and add it into UI / UX small improvements doc.
> > [Latest Build](https://drive.google.com/file/d/1fUlL_SPutuPuqz3Q3B6QDAZHZJjVSkJe/view?usp=sharing)
> 
> Thanks @dmitrii-gaico. Can you check the latest build?

Fixed in the latest 18408 Build. Thank you!

miorsufianiohk commented:
Hi @daniloprates , findings on 18408:

I am finding it a bit strange when the reward amounts are not right justify. It makes more sense to right justify as that's how most accounting system works. [See screenshot](https://drive.google.com/file/d/1FDt-TNxWgZb1jQcNlUx7mkSILgqXtWIU/view?usp=sharing)

gabriela-ponce commented:
@daniloprates The latest fixes look good. Issue summary is updated 👍

dmitrii-gaico commented:
> Hi @daniloprates , findings on 18408:
> 
> I am finding it a bit strange when the reward amounts are not right justify. It makes more sense to right justify as that's how most accounting system works. [See screenshot](https://drive.google.com/file/d/1FDt-TNxWgZb1jQcNlUx7mkSILgqXtWIU/view?usp=sharing)

I believe we need Styling Master @alexander-rukin here to take a look.

alexander-rukin commented:
> Hi @daniloprates , findings on 18408:
> 
> I am finding it a bit strange when the reward amounts are not right justify. It makes more sense to right justify as that's how most accounting system works. [See screenshot](https://drive.google.com/file/d/1FDt-TNxWgZb1jQcNlUx7mkSILgqXtWIU/view?usp=sharing)

right align -> this is a way

alexander-rukin commented:
Need to fix this ".000000"

daniloprates commented:
@gabriela-ponce @miorsufianiohk  @dmitrii-gaico @alexander-rukin here's what was done today:

* Correct reward amount formatting (we were printing an extra million ada)
* Revert shadow (remove conditional shadow)
* Amount right aligned
* General table styling adjustments
* Japanese translation for `Stake pool details are unavailable for this stake pool ID.`

miorsufianiohk commented:
Hi @daniloprates, on build 18430

The reward amount is still not right justified. Please [see screenshot](https://drive.google.com/file/d/1fsWVMyI-jMddBPIu9nNRiCkl91nQ5tBK/view?usp=sharing)

alexander-rukin commented:
> Hi @daniloprates, on build 18430
> 
> The reward amount is still not right justified. Please [see screenshot](https://drive.google.com/file/d/1fsWVMyI-jMddBPIu9nNRiCkl91nQ5tBK/view?usp=sharing)

this is because of a "Details" button on the right :( 
in the case of right alignment here, it will look awkward ((

alexander-rukin commented:
shouldn't we show scroll bar on the right like we doo for stake pool list?

<img width="698" alt="image" src="https://user-images.githubusercontent.com/2539345/123771064-6fc7db80-d8d3-11eb-8a0f-d1d749a78eda.png">

here how it's now (with no scroll table it's even more ridiculous) 

<img width="727" alt="image" src="https://user-images.githubusercontent.com/2539345/123771121-7b1b0700-d8d3-11eb-9c08-8d6b8f0062e8.png">

dmitrii-gaico commented:
Empty Details table is showed even Wallet has rewards. No loading spinner displayed.
[Pic0](https://drive.google.com/file/d/15zLvOVg2TVdUfB7Eem71fPiYAAfVMgxB/view?usp=sharing)
[Pic1](https://drive.google.com/file/d/1NHv3WqqVaIaBJxPRD5-vBtBR_i4HdDRv/view?usp=sharing)

gabriela-ponce commented:
@daniloprates Regarding the issue @dmitrii-gaico mentioned [here](https://github.com/input-output-hk/daedalus/pull/2572#issuecomment-870427402), we found the following:
- The issue only reproduces in one specific wallet, on all platforms. (The wallet named "Shelley" from [this set](https://drive.google.com/file/d/1IWuEJRTLl-KVn91wPpSd_Pn70kc_ZDGp/view?usp=sharing))
- There is a console error when clicking on "Details" for this wallet. Check [this](https://drive.google.com/file/d/1IaJax68k27IkZEFYWjehZhboIFO67wXu/view?usp=sharing) screenshot.
- We couldn't reproduce this issue on other wallets. Check [this](https://drive.google.com/file/d/1IneQP-K-fKIEth-2zqNi9gEgEZoHlmal/view?usp=sharing) screenshot from "TestWallet" on the same set.

daniloprates commented:
> Hi @daniloprates, on build 18430
> 
> The reward amount is still not right justified. Please [see screenshot](https://drive.google.com/file/d/1fsWVMyI-jMddBPIu9nNRiCkl91nQ5tBK/view?usp=sharing)

@miorsufianiohk could you check in the latest build?

![image](https://user-images.githubusercontent.com/1504716/124003095-ea185e80-d9ac-11eb-9f75-bb35ea1bb19a.png)

daniloprates commented:
> shouldn't we show scroll bar on the right like we doo for stake pool list?
> 

@alexander-rukin not sure I get what to do. Should I go beyond the Dialog padding?

![image](https://user-images.githubusercontent.com/1504716/124004188-28624d80-d9ae-11eb-9f9e-4b93996a7dc4.png)

daniloprates commented:
@gabriela-ponce @miorsufianiohk @dmitrii-gaico @alexander-rukin 

I think all styling issues are fixed.

I've added an error handling and am waiting for Darko for copy & translation. Also reached out to Rhys to check if we will ever have this error happening on mainnet.

![image](https://user-images.githubusercontent.com/1504716/124009809-6b272400-d9b4-11eb-968f-ae82e7d21944.png)

miorsufianiohk commented:
Hi @daniloprates ,

I was referring to this one. [See screenshot](https://drive.google.com/file/d/1avC-lfNSDmyjbEJjSCC8Ceg63BdMJTJv/view?usp=sharing). Tested again on 18463

daniloprates commented:
> Hi @daniloprates ,
> 
> I was referring to this one. [See screenshot](https://drive.google.com/file/d/1avC-lfNSDmyjbEJjSCC8Ceg63BdMJTJv/view?usp=sharing). Tested again on 18463

@mior in this screen the amount is not right-aligned, because of the details button

![image](https://user-images.githubusercontent.com/1504716/124140211-aaaa4a80-da5e-11eb-8dc0-69fb9e4199b3.png)

miorsufianiohk commented:
Hi @daniloprates , thanks for the reply.

 I am scrutinising from consistency point of view as the amount in Reward Details ares right-aligned. The amounts in Rewards screen will look more professional and appropriate if they are displayed like in the Rewards Details. 

![Screenshot 2021-07-01 at 16 21 09](https://user-images.githubusercontent.com/67595081/124150398-71d39b00-da89-11eb-9196-4b5615398c02.png)

Looking at the code, I see that the div for button "DETAILS" sits within the div for the amount. Is it not possible to create another container so both divs are separate?, ie 
(div)
 (div)
 Amount
  (/div)
(div)
 Button
  (/div)
(/div)

I might be wrong, but I think by having this, you could then right justify the amount. Any thought? 🤔 

> @Mior in this screen the amount is not right-aligned, because of the details button

daniloprates commented:
> Hi @daniloprates , thanks for the reply.
> 
> I am scrutinising from consistency point of view as the amount in Reward Details are right-aligned. The amounts in Rewards screen will look more professional and appropriate if they are displayed like in the Rewards Details.
> 
> ![Screenshot 2021-07-01 at 16 21 09](https://user-images.githubusercontent.com/67595081/124150398-71d39b00-da89-11eb-9196-4b5615398c02.png)
> 
> Looking at the code, I see that the div for button "DETAILS" sits within the div for the amount. Is it not possible to create another container so both divs are separate?, ie
> (div)
> (div)
> Amount
> (/div)
> (div)
> Button
> (/div)
> (/div)
> 
> I might be wrong, but I think by having this, you could then right justify the amount. Any thought? 🤔
> 
> > @Mior in this screen the amount is not right-aligned, because of the details button

cc @alexander-rukin ☝️

alexander-rukin commented:
> Hi @daniloprates , thanks for the reply.
> 
> I am scrutinising from consistency point of view as the amount in Reward Details are right-aligned. The amounts in Rewards screen will look more professional and appropriate if they are displayed like in the Rewards Details.
> 
> ![Screenshot 2021-07-01 at 16 21 09](https://user-images.githubusercontent.com/67595081/124150398-71d39b00-da89-11eb-9196-4b5615398c02.png)
> 
> Looking at the code, I see that the div for button "DETAILS" sits within the div for the amount. Is it not possible to create another container so both divs are separate?, ie
> (div)
> (div)
> Amount
> (/div)
> (div)
> Button
> (/div)
> (/div)
> 
> I might be wrong, but I think by having this, you could then right justify the amount. Any thought? 🤔
> 
> > @Mior in this screen the amount is not right-aligned, because of the details button

<img width="817" alt="image" src="https://user-images.githubusercontent.com/2539345/124155275-fe3c8880-da9e-11eb-93d1-845e38deac83.png">

right alignment is just not an option here, where we should reserve this right part of a table for "details" button, which is shown only on hover

alexander-rukin reviewed (approved):
Great job!

daniloprates commented:
@dmitrii-gaico @gabriela-ponce the "no data" issue is not happening anymore to that wallet. As Rhys said: 

> The deployment is managed by the ops team so I can't say definitively, but this could happen if the database was being re-synced. The mainnet instances will not be impacted by this condition as the database rebuild will now take place offline

But anyway, I've added a handling for that, it can be tested [in the storybook](https://user-images.githubusercontent.com/1504716/124287077-fd9b0500-db25-11eb-83b5-c06f25526c32.png).

daniloprates commented:
> <img alt="image" width="475" src="https://user-images.githubusercontent.com/2539345/122928329-6ccc6880-d372-11eb-8717-46f78a74d135.png">
> 
> is it ok to have this top line in grey? and not in a rank color (now we are changing this, but since change is not live...)

@alexander-rukin ranking here is not desirable, since we're displaying history data.

alexander-rukin commented:
> > <img alt="image" width="475" src="https://user-images.githubusercontent.com/2539345/122928329-6ccc6880-d372-11eb-8717-46f78a74d135.png">
> > is it ok to have this top line in grey? and not in a rank color (now we are changing this, but since change is not live...)
> 
> @alexander-rukin ranking here is not desirable, since we're displaying history data.

We can remove it here than. But not a big issue I think.
mb changing label to "current rank" is more appropriate

gabriela-ponce commented:
@daniloprates the latest fixes look good. I was able to check the [error message](https://drive.google.com/file/d/1S8szJmM3Bm02MTrN4latNEQC4pS-2anP/view?usp=sharing) by disconnecting the Internet before fetching the rewards, and it looks ok in my opinion.
Also, the rewards are visible now for the wallet that @dmitrii-gaico [mentioned](https://github.com/input-output-hk/daedalus/pull/2572#issuecomment-870427402).

However, there is a small issue with the same wallet. I'm guessing this happens because the grid has only one row, but there is a lot of space at the bottom of it. Check [this](https://drive.google.com/file/d/1bWozT6jTMQesErdQayi3d9bwBaE3kiJi/view?usp=sharing) screenshot.

Updating the issue summary ⏳

alexander-rukin commented:
> @daniloprates the latest fixes look good. I was able to check the [error message](https://drive.google.com/file/d/1S8szJmM3Bm02MTrN4latNEQC4pS-2anP/view?usp=sharing) by disconnecting the Internet before fetching the rewards, and it looks ok in my opinion.

@daniloprates what about one button for those cases? https://zpl.io/2GD3JzE - Darko liked this one if I'm not mistaken

daniloprates commented:
@gabriela-ponce @miorsufianiohk @dmitrii-gaico I've finished implementing the [actions column](https://app.zeplin.io/project/57acfaa7436a750316161a8b/screen/5a159c345400b2aa34b7ff8a) Mior suggested and the last issues. Could you guys give it another go?

alexander-rukin commented:
<img width="170" alt="image" src="https://user-images.githubusercontent.com/2539345/124567263-824e9180-de4c-11eb-96dd-21bbd5e51530.png">

@daniloprates on hover we shouldn't change opacity for selected column icon (downgrading opacity level looks strange)

alexander-rukin commented:
<img width="310" alt="image" src="https://user-images.githubusercontent.com/2539345/124567418-a7db9b00-de4c-11eb-9341-676290d87177.png">

I'm able to click this column header and have sorting there 😱

aleksandardjordjeviciohk commented:
@daniloprates @DominikGuzei i have rechecked functionalities and noticed one thing with the scroll list. Sometimes scrollbar get stuck and starts cutting while you're scrolling. Pls recheck video 

https://user-images.githubusercontent.com/15862062/124568142-ea00de80-de44-11eb-8b60-76689853f5a8.mov

dmitrii-gaico commented:
@daniloprates Hi Danilo. Some small issues below :  
1. We should not display the scroll bar if list does fits with no hassle. [Pic1](https://drive.google.com/file/d/1lsGMGtJIZc-_1xEjHVw-KCc3ZXmSVKtT/view?usp=sharing) [Pic2](https://drive.google.com/file/d/1OGbiFKKloZPCmBbGFOkIDYMKX0rpzbla/view?usp=sharing)
2. Can we make Headers "Wallet \ Rewards address \ Total (ADA)" Non Selectable ? The issue is when user is sorting couple of time the same header the Text is selected. This looks inconsistent. [Record](https://drive.google.com/file/d/1UxPYJAqK5LUm_nfe6zkLsV9apLCcAWUE/view?usp=sharing)

daniloprates commented:
@alexander-rukin @dmitrii-gaico thanks, they're all fixed.

@aleksandardjordjeviciohk I couldn't reproduce this. It seems to be some processing issues, maybe your machine was overloaded? It doesn't seem to be something we can change, unless we go for a virtual list.

alexander-rukin reviewed (approved):
All good now :)

miorsufianiohk commented:
Hi @daniloprates , here is the report on build 18509.

I found the pool popup in "Reward Details" displaying different data than pool popup in "Stake Pools screen". For example in [Reward Details view](https://drive.google.com/file/d/19cytK0WkX_UlZCfLyTByG6MskguikwuD/view?usp=sharing), the Ranking for pool ANP is displayed as "1*", whereas in [Stake Pools view](https://drive.google.com/file/d/1zc-QBUFIplfXQWejbntIlUJcl-uImJF4/view?usp=sharing), the Ranking for pool ANP is displayed as "321*".

nikolaglumac commented:
@daniloprates please work closely with @miorsufianiohk to fix the remaining issues today 🙏

daniloprates commented:
> * I found the pool popup in "Reward Details" displaying different data than pool popup in "Stake Pools screen". For example in [Reward Details view](https://drive.google.com/file/d/19cytK0WkX_UlZCfLyTByG6MskguikwuD/view?usp=sharing), the Rank for pool ANP is displayed as "1*", whereas in [Stake Pools view](https://drive.google.com/file/d/1zc-QBUFIplfXQWejbntIlUJcl-uImJF4/view?usp=sharing), the Rank for pool ANP is displayed as "321*".
> * Also, using KIWI pool as an example, the colour of the Rank in this [Reward Details view](https://drive.google.com/file/d/1oDeJskDBC-_nmk6DSEVsMOYRMbCSIGBs/view?usp=sharing) , is different from [Stake Pools view](https://drive.google.com/file/d/1BVDWMZBWmqlPBTsXc4JxPkL6j83cqx9d/view?usp=sharing)

@miorsufianiohk thanks, nice catch. The ranking here is irrelevant, so it [was removed](https://user-images.githubusercontent.com/1504716/124923855-3a0cac00-dfd1-11eb-9923-ae7804a128c5.png). The grey colorband meant "no ranking" but it was confusing, so was removed as well.

> * Additionally, on design, the text in column header is [Total Rewards Earned (ADA) ](https://drive.google.com/file/d/18EmSXRm9asJYR24PuZOczoyqTfybYkrb/view?usp=sharing)whereas in app it's [Total (ADA)](https://drive.google.com/file/d/1V1mTukVMFgKFGXjlRp_bTXeRuAA1gu-E/view?usp=sharing)


The column [was renamed](https://input-output-rnd.slack.com/archives/G0186R8BWNR/p1625257477073500?thread_ts=1625231323.070400&cid=G0186R8BWNR).

miorsufianiohk commented:
Hi @daniloprates

Tested on 18542.
Thanks for addressing these[ issues](https://github.com/input-output-hk/daedalus/pull/2572#issuecomment-876299779) .
Everything looks good IMO. Great work and thank you 👍 

I am unable to test the issue found by @gabriela-ponce  as I don't have Reward Details with only one row.

nikolaglumac commented:
@daniloprates is anyone working on the Netlify issue?

daniloprates commented:
> @daniloprates is anyone working on the Netlify issue?

@nikolaglumac nope. @DominikGuzei could you have a look on this? Or grant me access, so I can check?

gabriela-ponce commented:
@daniloprates The fixes look good! I wasn't able to check the issue with the single-row on rewards, as the wallet I used now has more items on the list.

There is a small issue with the scrollbar that I was able to reproduce on macOS: The list moves, but there is a delay in the scrollbar. Check [this](https://drive.google.com/file/d/1q_QvXC5pdS8stjTZr5FWxdrnKBczVd48/view?usp=sharing) video.

daniloprates commented:
> @daniloprates The fixes look good! I wasn't able to check the issue with the single-row on rewards, as the wallet I used now has more items on the list.
> 
> There is a small issue with the scrollbar that I was able to reproduce on macOS: The list moves, but there is a delay in the scrollbar. Check [this](https://drive.google.com/file/d/1q_QvXC5pdS8stjTZr5FWxdrnKBczVd48/view?usp=sharing) video.

@gabriela-ponce, @aleksandardjordjeviciohk also [reported](https://github.com/input-output-hk/daedalus/pull/2572#issuecomment-874567411) this scrolling issue, but I could never reproduce it. Is there any extra steps for it to happen? 

What are you machine, guys? Macbook pro?

gabriela-ponce commented:
> @gabriela-ponce, @aleksandardjordjeviciohk also [reported](https://github.com/input-output-hk/daedalus/pull/2572#issuecomment-874567411) this scrolling issue, but I could never reproduce it. Is there any extra steps for it to happen?
> 
> What are you machine, guys? Macbook pro?

@daniloprates My bad, I missed that comment! I'll check the issues and update the summary so we're not missing anything. 
I can reproduce this issue in MacBook Pro 2019, BigSur.

daniloprates commented:
@gabriela-ponce I've updated the [Storybook story](http://localhost:6006/?path=/story/decentralization-staking--rewards-history), adding a slider for the number of items, so you can check the 1 item scenario, also multiple items. Could you check if the scrolling issue happens there as well?

(The Netifly build is broken, so you need to do it locally, by running `yarn run storybook`)

gabriela-ponce commented:
@daniloprates I've updated the list of the issues, everything looks good.

This is the only pending issue:
- When adding up the values, sometimes the result doesn't match with the total on the rewards table. On build 18585 (that contains the additional logs) there are 3 different scenarios:
  - The sum of the values on the list matches the total. Check [this screenshot](https://drive.google.com/file/d/1_3jVhP2jaU9tJaQwxgaULPQyLtZmIBPr/view?usp=sharing).
  - The sum doesn't match and it's because some epoch [is omitted from the total.](https://drive.google.com/file/d/1oSUtF0QXIC248kvDgbUyTEudWSV6RmhF/view?usp=sharing) Later, on the same build and same wallet, [the total can be correct](https://drive.google.com/file/d/1XAf1vz5aReFOVwTHtGyEuzIjo4IOPzbV/view?usp=sharing). 
  - The sum doesn't match and we don't know if some epoch was omitted from the total. In these cases, the total can be higher [on the initial screen and smaller on the details.](https://drive.google.com/file/d/1addr4AdETwj5O3XwKQyknimlReXIJuHa/view?usp=sharing)

nikolaglumac commented:
This PR is paused/blocked due to cardano-db-sync issue. We need to wait for it to be fixed on their side first: https://github.com/input-output-hk/cardano-db-sync/issues/696

cc @daniloprates @darko-mijic @rhyslbw

simonmichael commented:
Thanks for the work on this. Could the date the reward was awarded be added to the dialog and (especially) the CSV ? That will be informative and will make this data much more consumable by accounting software.

nikolaglumac commented:
> Thanks for the work on this. Could the date the reward was awarded/earned be added to the dialog and (especially) the CSV ? That will be informative to the human reader, will make this data much more consumable by accounting software, and could be important for calculating capital gains.

@daniloprates can you please check if this is possible? I am unsure if we get the date of the reward... It could be that we get only epoch number... We could perhaps use the start of the epoch as the date perhaps?
cc @darko-mijic

daniloprates commented:
> > Thanks for the work on this. Could the date the reward was awarded/earned be added to the dialog and (especially) the CSV ? That will be informative to the human reader, will make this data much more consumable by accounting software, and could be important for calculating capital gains.
> 
> @daniloprates can you please check if this is possible? I am unsure if we get the date of the reward... It could be that we get only epoch number... We could perhaps use the start of the epoch as the date perhaps?
> cc @darko-mijic

You are correct, @nikolaglumac, the only information we receive is `address`, `amount`, `epoch` and `Stake Pool`

nikolaglumac commented:
> You are correct, @nikolaglumac, the only information we receive is `address`, `amount`, `epoch` and `Stake Pool`

Perhaps we could display the start of that epoch as the rewards pay-out time?

miorsufianiohk commented:
Hi @daniloprates ,

On build 19988 Mainnet:

We found out that some of the wallets were not able to display the reward details screen ([see screenshot which contains error on console](https://drive.google.com/file/d/1MOfgPQaoK_fXwTIWvQcnDxFzI3tj9i1L/view?usp=sharing)), whereas some were able to ([see screenshot](https://drive.google.com/file/d/14cu2ULKnIWXDUxbCmlF3Ab2UqxJXIwaC/view?usp=sharing)).

DominikGuzei commented:
@daniloprates any chance we can calculate a date from the epoch? I just used it for my own tax calculations and without dates its much less useful

daniloprates commented:
@DominikGuzei sure thing. We've gained some time since it was not released. Can you coordinate so it gets done by EOY?

mchappell commented:
Is this WIP/On-hold? If so can we update the relevant labels.

DominikGuzei commented:
> Is this WIP/On-hold? If so can we update the relevant labels.

Yes, sorry this is still WIP
